x-pack/plugins/telemetry/common/check_deprecated.ts

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { KibanaConfig } from 'src/legacy/server/kbn_server';
+
+export function checkDeprecated(config: KibanaConfig, configPath: string) {
+  const deprecatedConfig = config.get(`xpack.xpack_main.${configPath}`);
+  if (typeof deprecatedConfig !== 'undefined') {
```

Comment:
curious if this is TS thing instead of using `if (deprecatedConfig !== undefined)` instead.

Reply:
Should this function be logging a message if it finds a deprecated setting?  Maybe something else in Kibana does that?

The name of the module / function seem a bit off; the purpose is to get a config value, checking for deprecated is just some extra goodness.  So it reads weird in usage, alongside `config.get()`, for example.  Something like `configGet()` or if the wordiness is acceptable (not sure!) `configGetWithDeprecated()` or such.

Reply:
Just an old habit. Before es6 `typeof` was safe, it never threw a `ReferenceError` even if the variable is not initialized. Now it will throw if the variable is in the [TDZ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let#Temporal_Dead_Zone) so it is not as safe as it used to be, but I still prefer it over direct checking.


```
// Case 1:
typeof a; // ReferenceError
let a = 1; 

// Case 2:
typeof b; // undefined

// Case 3:
console.log(b) // ReferenceError
```

So `typeof` just avoid the third case, which is caught by TS at compile time  so it's not an issue. TS (and myself) don't mind using neither :)

Reply:
Ah, I see code below warning about the deprecations ...

Reply:
Yes `deprecations` method in the `xpack_main` plugin warns about deprecations. Changing the name does make sense. Thank you for the thorough review!